ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -443,11 +443,11 @@ void execute_program(std::string operation, std::string fname, std::string args,
 void execute_program(std::string fname, std::string args, bool wait) { execute_program("open", fname, args, wait); }
 
 std::string environment_get_variable(std::string name) {
-  WCHAR buffer[1024];
+  WCHAR buffer[SHRT_MAX];
```

Comment:
This is begging for a stack overflow. You are allocating 128 KB onto a 1MB stack. This variable alone occupies an eighth of stack memory.

Reply:
@JoshDreamland i did that because @RobertBColton requested me to. I'll revert it